src/utils/wallet.js

```diff
@@ -191,7 +315,16 @@ class Wallet {
     }
 
     async removeAccessKey(publicKey) {
-        return await this.getAccount(this.accountId).deleteKey(publicKey)
+        const { account, has2fa } = await this.getAccountAndState()
+        if (has2fa) {
+            const request = {
+                receiver_id: this.accountId,
+                actions: [{ type: 'DeleteKey', public_key: splitPK(publicKey) }]
```

Comment:
seems like it's better to make contract handle `ed25519:` prefix as well to make it compatible with all other places using string representation of key

Reply:
not sure it's possible atm due to borsh/rust sdk types. Will inquire.